controllers/metal3.io/host_state_machine_test.go

```diff
@@ -146,6 +146,278 @@ func TestProvisioningCapacity(t *testing.T) {
 	}
 }
 
+func TestDetach(t *testing.T) {
+	testCases := []struct {
+		Scenario                  string
+		Host                      *metal3v1alpha1.BareMetalHost
+		Annotations               map[string]string
+		ExpectedDetach            bool
```

Comment:
I think this is a little bit misleading for how it has been defined (see also my other comment in the mock). Moreover, I'd suggest to remove it, as I think it should be sufficient to verify the host operationalStatus/state and action result

Reply:
Proving the OperationalStatus is set isn't exactly the same as proving the provisioner.Detach() was actually called, but I can remove this if we don't think that coverage is important.

Reply:
I converted this to a generic method that proves if the provisioner method got called or not